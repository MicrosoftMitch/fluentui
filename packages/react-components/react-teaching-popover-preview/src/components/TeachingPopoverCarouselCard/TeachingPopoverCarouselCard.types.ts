import { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type TeachingPopoverCarouselCardSlots = {
  /**
   * The element wrapping the buttons.
   */
  root: NonNullable<Slot<'div'>>;
};

export type TeachingPopoverCarouselCardProps = ComponentProps<TeachingPopoverCarouselCardSlots> & {
  /* Required: Passed into CarouselItem to identify pages. */
  value: string;
};

export type TeachingPopoverCarouselCardState = ComponentState<TeachingPopoverCarouselCardSlots> &
  Required<Pick<TeachingPopoverCarouselCardProps, 'value'>>;
