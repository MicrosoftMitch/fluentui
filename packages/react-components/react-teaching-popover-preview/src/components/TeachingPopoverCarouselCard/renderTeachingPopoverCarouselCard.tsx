/** @jsxRuntime automatic */
/** @jsxImportSource @fluentui/react-jsx-runtime */
import { CarouselItem_unstable } from '../TeachingPopoverCarousel/Carousel/CarouselItem';
import type { TeachingPopoverCarouselCardState } from './TeachingPopoverCarouselCard.types';
import { TeachingPopoverCarouselCardSlots } from './TeachingPopoverCarouselCard.types';
import { assertSlots } from '@fluentui/react-utilities';

/**
 * Render the final JSX of TeachingPopoverCarouselCard
 */
export const renderTeachingPopoverCarouselCard_unstable = (state: TeachingPopoverCarouselCardState) => {
  assertSlots<TeachingPopoverCarouselCardSlots>(state);

  return CarouselItem_unstable({ children: <state.root />, value: state.value });
};
