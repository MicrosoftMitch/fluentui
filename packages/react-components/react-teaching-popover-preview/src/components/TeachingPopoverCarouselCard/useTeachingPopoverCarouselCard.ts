import * as React from 'react';
import { getIntrinsicElementProps, slot } from '@fluentui/react-utilities';
import type {
  TeachingPopoverCarouselCardProps,
  TeachingPopoverCarouselCardState,
} from './TeachingPopoverCarouselCard.types';
/**
 * Returns the props and state required to render the component
 * @param props - TeachingPopoverCarouselCard properties
 * @param ref - reference to root HTMLElement of TeachingPopoverCarouselCard
 */
export const useTeachingPopoverCarouselCard_unstable = (
  props: TeachingPopoverCarouselCardProps,
  ref: React.Ref<HTMLDivElement>,
): TeachingPopoverCarouselCardState => {
  const { value } = props;

  return {
    components: {
      root: 'div',
    },
    root: slot.always(
      getIntrinsicElementProps('div', {
        ref,
        ...props,
      }),
      { elementType: 'div' },
    ),
    value,
  };
};
