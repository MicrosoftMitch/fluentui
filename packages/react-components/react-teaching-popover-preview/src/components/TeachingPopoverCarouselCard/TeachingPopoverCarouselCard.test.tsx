import * as React from 'react';
import { render } from '@testing-library/react';
import { isConformant } from '../../testing/isConformant';
import { TeachingPopoverCarouselCard } from './TeachingPopoverCarouselCard';
import { resetIdsForTests } from '@fluentui/react-utilities';
import { mockTeachingPopoverCarouselContext } from '../../testing/mockTeachingPopoverCarouselContext';

describe('TeachingPopoverCarouselCard', () => {
  isConformant({
    Component: TeachingPopoverCarouselCard,
    displayName: 'TeachingPopoverCarouselCard',
    requiredProps: { value: 'test' },
  });

  beforeEach(() => {
    resetIdsForTests();
    mockTeachingPopoverCarouselContext({});
  });

  it('renders a default state', () => {
    const result = render(
      <TeachingPopoverCarouselCard value="test">Default TeachingPopoverCarouselCard</TeachingPopoverCarouselCard>,
    );
    expect(result.container).toMatchSnapshot();
  });
});
