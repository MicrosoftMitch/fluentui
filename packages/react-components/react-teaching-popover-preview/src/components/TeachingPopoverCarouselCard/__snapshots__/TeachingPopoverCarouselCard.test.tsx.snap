// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeachingPopoverCarouselCard renders a default state 1`] = `
<div>
  <span
    data-carousel-active-item="false"
    data-carousel-item="test"
    hidden=""
  />
</div>
`;
